tests/std/tests/GH_000935_complex_numerical_accuracy/floating_point_utils.hpp

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <cassert>
+#include <cfenv>
+#include <cmath>
+#include <float.h>
+#include <type_traits>
+#include <xutility>
+
+namespace fputil {
+    template <typename T>
+    using float_bits_t = typename _STD _Float_traits<T>::type;
+
+    template <typename T>
+    _INLINE_VAR constexpr float_bits_t<T> magnitude_mask_v = _STD _Float_traits<T>::_Magnitude_mask;
+
+    template <typename T>
+    _INLINE_VAR constexpr float_bits_t<T> exponent_mask_v = _STD _Float_traits<T>::_Exponent_mask;
+
+    template <typename T>
+    _INLINE_VAR constexpr float_bits_t<T> significand_mask_v = magnitude_mask_v<T> & ~exponent_mask_v<T>;
+
+    template <typename T>
+    _INLINE_VAR constexpr float_bits_t<T> sign_mask_v = _STD _Float_traits<T>::_Sign_mask;
+
+    template <typename T>
+    _INLINE_VAR constexpr float_bits_t<T> norm_min_bits_v = significand_mask_v<T> + 1U;
+
+    template <typename T>
+    _INLINE_VAR constexpr float_bits_t<T> norm_max_bits_v = exponent_mask_v<T> - 1U;
+
+    template <typename T>
+    _INLINE_VAR constexpr float_bits_t<T> infinity_bits_v = exponent_mask_v<T>;
+
+    // not affected by abrupt underflow
+    template <typename T, std::enable_if_t<std::is_floating_point<T>::value, int> = 0>

```

Comment:
We conventionally use the variable templates for type traits (which we implemented unconditionally, even in C++14 mode). They're less verbose and more efficient for MSVC to compile, and I claim that this convention has nothing to do with the fact that I proposed them for Standardization. :joy_cat:

Trying to avoid unnecessary style nitpicks in test code, but I think it's worth changing these occurrences - it makes searching product code and test code simultaneously somewhat easier if we want to find all functions constrained by `is_floating_point_v`, for example.